packages/react-components/react-virtualizer/src/hooks/useVirtualizerMeasure.ts

```diff
@@ -0,0 +1,102 @@
+import * as React from 'react';
+import { canUseDOM } from '@fluentui/react-utilities';
+import { useFluent_unstable as useFluent } from '@fluentui/react-shared-contexts';
+
+/**
+ * React hook that measures virtualized space based on a static size to ensure optimized virtualization length.
+ */
+export const useStaticVirtualizerMeasure = (
+  defaultItemSize: number,
+  scrollView: HTMLElement | null = null,
+  direction: 'vertical' | 'horizontal' = 'vertical',
+): {
+  virtualizerLength: number;
+  bufferItems: number;
+  bufferSize: number;
+} => {
+  const [virtualizerLength, setVirtualizerLength] = React.useState(0);
+  const [virtualizerBufferItems, setVirtualizerBufferItems] = React.useState(0);
+  const [virtualizerBufferSize, setVirtualizerBufferSize] = React.useState(0);
+
+  const { targetDocument } = useFluent();
+
+  // If no ref is provided, the virtualizer should be in the main body document's scroll.
```

Comment:
I don't believe we should fallback to document.body, we should make a user set element necessary, but ofc it should still work if that user set element is document.body